tests/unittest/_torch/visual_gen/test_wan_vae.py

```
@@ -33,6 +51,388 @@
+        dynamic_activation_quant=False,
+    )
+
+    assert loaded is model
+    assert not any(isinstance(module, NVFP4WanCausalConv3d) for module in model.modules())
+
+
+@pytest.mark.parametrize("checkpoint_is_fp4", [False, True])
+def test_dynamic_weight_request_must_match_checkpoint_source(checkpoint_is_fp4):
+    incompatible_request = checkpoint_is_fp4
+
+    with pytest.raises(ValueError, match="weights.dynamic"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)